### DISCUSSION TITLE
Expand macros (not recursivly)

### DISCUSSION THREAD

**Discussion by TomzBench** [mrcjkb/rustaceanvim#257] [Q&A] [answered]
Is there a way to expand macros in a not recursive manor? 

Usually when I expanding macros, if there is any recursion it has been third party macros that I dont care about peeking at. It's actually easier to debug when its not recursive. Though It is valuable to recursivly expand macros also - I typically prefer not to. It would be nice to have an option to only expand macros without recursion.

**Comment by mrcjkb** [comment, marked as answer]
Hey :wave:

macro expansion is handled by rust-analyzer.
Afaik, [it's always recursive](https://rust-analyzer.github.io/manual.html#expand-macro-recursively).

You'll probably want to follow [this issue](https://github.com/rust-lang/rust-analyzer/issues/11888).

**Reply by TomzBench** [reply]
ahh i see. thanks!
